Failure in WatcherRestartIT.testWatcherRestart

**Build scan**: https://gradle-enterprise.elastic.co/s/pgbuawfavusl2

**Repro line**: `./gradlew ':x-pack:qa:rolling-upgrade:v7.9.1#oneThirdUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.WatcherRestartIT" -Dtests.method="testWatcherRestart" -Dtests.seed=3DE603A8143C17BA -Dtests.security.manager=true -Dtests.bwc=true -Dtests.locale=fr-CA -Dtests.timezone=Africa/Tripoli -Druntime.java=8	
`

**Reproduces locally?**: No

**Applicable branches**: 7.x, 7.11, 7.12

**Failure history**: Failing pretty frequently as of the morning of 3/3/21

**Failure excerpt**:

```

java.lang.AssertionError: |  
-- | --
  | Expected: not a string containing "\"watcher_state\":\"stopped\"" |  
  | but: was "{\"_nodes\":{\"total\":3,\"successful\":3,\"failed\":0},\"cluster_name\":\"v7.9.1\",\"manually_stopped\":false,\"stats\":[{\"node_id\":\"qEEjrtXgSVqMj7QTR-ec6w\",\"watcher_state\":\"started\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":0}},{\"node_id\":\"zxGWpG4yRSO_e2B8k2Xyqw\",\"watcher_state\":\"stopped\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":0}},{\"node_id\":\"orv9jd8cTkOqJNipcPYfNg\",\"watcher_state\":\"started\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":1}}]}"


at __randomizedtesting.SeedInfo.seed([3DE603A8143C17BA:A8BE0D275F20CEE3]:0) |  
-- | --
  |   | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  |   | at org.junit.Assert.assertThat(Assert.java:956) |  
  |   | at org.junit.Assert.assertThat(Assert.java:923) |  
  |   | at org.elasticsearch.upgrades.WatcherRestartIT.lambda$ensureWatcherStarted$2(WatcherRestartIT.java:153) |  
  |   | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1014) |  
  |   | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:987) |  
  |   | at org.elasticsearch.upgrades.WatcherRestartIT.ensureWatcherStarted(WatcherRestartIT.java:147) |  
  |   | at org.elasticsearch.upgrades.WatcherRestartIT.testWatcherRestart(WatcherRestartIT.java:40)



```

There's a number of tests failing with this though, so it seems it might still be something generally watcher-esque.

```
java.lang.AssertionError: Failure in test setup: Failed to initialize at least 3 watcher nodes
```

This seems to be busted in both 7.x and 7.12 branches and specific to the BWC tests against the 7.9.x series.

--- a/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/support/WatcherIndexTemplateRegistry.java
+++ b/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/support/WatcherIndexTemplateRegistry.java
@@ -73,7 +73,7 @@ public class WatcherIndexTemplateRegistry extends IndexTemplateRegistry {
 
     @Override
     protected List<IndexTemplateConfig> getLegacyTemplateConfigs() {
-        if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_7_9_0)) {
+        if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_7_10_0)) {
             return Collections.emptyList();
         } else if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_7_7_0)) {
             return Collections.singletonList(
@@ -110,12 +110,10 @@ public class WatcherIndexTemplateRegistry extends IndexTemplateRegistry {
     }
 
     public static boolean validate(ClusterState state) {
-        final Stream<String> watcherHistoryTemplateIds;
-        if (state.nodes().getMinNodeVersion().onOrAfter(Version.V_7_9_0)){
-            watcherHistoryTemplateIds = state.getMetadata().templatesV2().keySet().stream();
-        } else {
-            watcherHistoryTemplateIds = Arrays.stream(state.getMetadata().getTemplates().keys().toArray(String.class));
-        }
+        // A .watch-history should exist, whether it is a legacy or composable index template
+        // that doesn't matter when deciding to start watcher.
+        final Stream<String> watcherHistoryTemplateIds = Stream.concat(state.getMetadata().templatesV2().keySet().stream(),
+            Arrays.stream(state.getMetadata().getTemplates().keys().toArray(String.class)));
         return watcherHistoryTemplateIds.filter(s -> s.startsWith(".watch-history-"))
             .map(s -> Integer.valueOf(s.substring(s.lastIndexOf('-') + 1)))
             .anyMatch(version -> version >= 9);

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/UpgradeClusterClientYamlTestSuiteIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/UpgradeClusterClientYamlTestSuiteIT.java
@@ -60,8 +60,7 @@ public class UpgradeClusterClientYamlTestSuiteIT extends ESClientYamlSuiteTestCa
                 }
             }, 1, TimeUnit.MINUTES);
         } catch (AssertionError e) {
-            // AwaitsFix: https://github.com/elastic/elasticsearch/issues/69918
-//            throw new AssertionError("Failure in test setup: Failed to initialize at least 3 watcher nodes", e);
+            throw new AssertionError("Failure in test setup: Failed to initialize at least 3 watcher nodes", e);
         }
     }
 

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/WatcherRestartIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/WatcherRestartIT.java
@@ -32,7 +32,6 @@ public class WatcherRestartIT extends AbstractUpgradeTestCase {
 
     private static final String templatePrefix = ".watch-history-";
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/69918")
     public void testWatcherRestart() throws Exception {
         client().performRequest(new Request("POST", "/_watcher/_stop"));
         ensureWatcherStopped();

